packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -25,113 +20,116 @@ export const getStyles = memoizeFunction((
   const checkboxBackgroundChecked = semanticColors.inputBackgroundChecked;
   const checkboxBackgroundCheckedHovered = semanticColors.inputBackgroundCheckedHovered;
   const checkboxBorderColorCheckedHovered = semanticColors.inputBackgroundCheckedHovered;
-  const checkboxBackgroundDisabled = semanticColors.disabledBodyText;
+  const checkboxBackgroundDisabledChecked = semanticColors.disabledBodyText;
   const checkboxTextColor = semanticColors.bodyText;
   const checkboxTextColorDisabled = semanticColors.disabledText;
 
-  const styles: ICheckboxStyles = {
+  return ({
     root: [
-      getFocusStyle(theme, -2),
+      'ms-Checkbox',
+      reversed && 'reversed',
+      checked && 'is-checked',
+      !disabled && 'is-enabled',
+      disabled && 'is-disabled',
+      getFocusStyle(theme, -3),
       theme.fonts.medium,
       {
         padding: '0',
         border: 'none',
         background: 'none',
-        backgroundColor: 'transparent',
         margin: '0',
         outline: 'none',
         display: 'block',
         cursor: 'pointer',
+      },
+      !disabled && [
+        !checked && {
+          selectors: {
+            ':hover .ms-Checkbox-checkbox': { borderColor: checkboxBorderHoveredColor },
+            ':focus .ms-Checkbox-checkbox': { borderColor: checkboxBorderHoveredColor }
+          }
+        },
+        checked && {
+          selectors: {
+            ':hover .ms-Checkbox-checkbox': {
+              background: checkboxBackgroundCheckedHovered,
+              borderColor: checkboxBorderColorCheckedHovered
+            },
+            ':focus .ms-Checkbox-checkbox': {
+              background: checkboxBackgroundCheckedHovered,
+              borderColor: checkboxBorderColorCheckedHovered
+            }
+          }
+        },
+        {
+          selectors: {
+            ':hover .ms-Checkbox-text': { color: palette.black },
+            ':focus .ms-Checkbox-text': { color: palette.black }

```

Comment:
Is there a reason we switch to black here? 
![image](https://user-images.githubusercontent.com/1434956/39595074-4d0d39c0-4ec3-11e8-870b-56813104ade8.png)

Reply:
I copied this change from commit c3d8e35ed. Should we use different colors instead?

Reply:
@micahgodbolt this would introduce getStyles, which would let you provide the right styling based on the state provided, and remove the overly verbose and limited styles list.

Reply:
I was just noting that in the original it was using a semantic color, and we swapped it out for palette. Not sure which is correct, but would want to make sure it was intentional.